tree/ntuple/v7/src/RNTupleModel.cxx

```diff
@@ -54,6 +54,21 @@ ROOT::RResult<void>
 ROOT::Internal::RProjectedFields::EnsureValidMapping(const ROOT::RFieldBase *target, const FieldMap_t &fieldMap)
 {
    auto source = fieldMap.at(target);
+
+   if (!target->GetColumnRepresentatives()[0].empty()) {
+      const auto representative = target->GetColumnRepresentatives()[0][0];
+      // If the user is trying to add a projected field upon which they called SetTruncated() or SetQuantized(),
+      // they probably have the wrong expectations about what should happen. Warn them about it.
+      if (representative == ENTupleColumnType::kReal32Trunc || representative == ENTupleColumnType::kReal32Quant) {
+         R__LOG_WARNING(ROOT::Internal::NTupleLog())
+            << "setting the column representative of a projected field to Real32Trunc or Real32Quant has no effect, as "
+               "the on-disk representation of the value is decided by the "
+               "projection source field. This will work, but the value range and bits of precision you set on the "
+               "projected field will be "
+               "ignored. You probably want to use a RField<float> or RField<double> as your projection target instead.";
```

Comment:
I'm not entirely sure how to interpret the suggestion here. Isn't the field type kind of orthogonal from the column representative in this case? I.e., I can make an `RField<float>` and call `SetQuantized` (as is done in the tests) on it, then if I get this warning suggesting me to use `RField<float>` seems a bit confusing.

Reply:
You are right, the message is misleading: I will reformulate it a bit better. It's still a `RField<float>` in all cases, but what needs to be said is that is makes no sense to call SetQuantized/Truncated on it.